src/bootstrap/compile.rs

```diff
@@ -67,6 +68,11 @@ impl Step for Std {
             return;
         }
 
+        // If this is stage 0 or 1 and we're using a cached rustc, copy the stage 0 standard library instead of rebuilding.
+        if std::env::var("BOOTSTRAP_CACHE_STAGE1").is_ok() && compiler.stage < 2 {
```

Comment:
This seems wrong - we should not need to read an environment variable to detect this; rustbuild should parse it out of the config.toml. (This is another reason not to try to do fancy detection of whether or not to do this).

Reply:
Ok, done.